robot/impl/resource_manager.go

```diff
@@ -138,7 +147,16 @@ func (manager *resourceManager) addRemote(
 			return
 		}
 	} else {
+		if manager.opts.ftdc != nil {
+			manager.opts.ftdc.Remove(rName.String())
+		}
+
+		// SwapResource can change the underlying resource from a non-remote to a remote?
```

Comment:
Uh lost again here; when would that happen?

Reply:
This was just me reasoning about the code. We're in `addRemote`, but there's this path about replacing. So I assumed we could go from this original config:
```
{
  components: [{name: "fakeMotor", type: "motor", model: "fake"}],
  remotes: []
}
```

to this interesting (read: antagonistic) config:
```
{
  components: [],
  remotes: [{name: "fakeMotor", address: "..."}]
}
```

Reply:
But maybe the name encodes the API + type so it'd be impossible to craft a "name conflict" in this way that confuses the resource manager/graph?

Reply:
> But maybe the name encodes the API + type so it'd be impossible to craft a "name conflict" in this way that confuses the resource manager/graph?

That is my understanding, yes. I don't think you could accidentally `SwapResource` and, in doing so, switch APIs because, as you mention, the resource name encodes the API + type.